### PR TITLE
Improve CI.

### PR DESCRIPTION
**Pull request by floitsch**

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -9,17 +9,17 @@ build/CMakeCache.txt:
...
+	cmake --build build --target check
 
 # We rebuild the cmake file all the time.
 # We use "glob" in the cmakefile, and wouldn't otherwise notice if a new
 # file (for example a test) was added or removed.
 # It takes <1s on Linux to run cmake, so it doesn't hurt to run it frequently.
 rebuild-cmake:
 	mkdir -p build
-	(cd build && cmake .. -G Ninja)
+	cmake -B build -DCMAKE_BUILD_TYPE=Debug
```
Comment (kasperl) [inline Makefile:23]: Why do we need to use Debug here?

File: tests/CMakeLists.txt [hunk 2]
```diff
@@ -24,7 +25,7 @@ math(EXPR PARALLEL_COUNT "2 * ${NUM_CPU}")
 
 add_custom_target(
   check
-  COMMAND ${CMAKE_CTEST_COMMAND} -j${PARALLEL_COUNT} -T test --output-on-failure
+  COMMAND ${CMAKE_CTEST_COMMAND} -j${PARALLEL_COUNT} -T test --output-on-failure -C Debug
```
Comment (kasperl) [inline tests/CMakeLists.txt:28]: Why do we need to use Debug here?

**Review by kasperl — APPROVED** [review]
Looks good to me.

File: Makefile (hunk 1 above)
Comment (floitsch) [reply Makefile:23]: The non-ninja builder requires a build-type. Otherwise the tests don't work on Windows.

File: tests/CMakeLists.txt (hunk 2 above)
Comment (floitsch) [reply tests/CMakeLists.txt:28]: The non-ninja builder requires a build-type. Otherwise the tests don't work on Windows.

**Pull request merged**